hasktorch/src/Torch/NN.hs

```diff
@@ -48,11 +51,36 @@ instance Parameterized Parameter where
   flattenParameters = pure
   replaceOwnParameters _ = nextParameter
 
+instance Parameterized a => Parameterized [a] where
+  flattenParameters [] = []
+  flattenParameters (x:xs) = flattenParameters x ++ flattenParameters xs
+  replaceOwnParameters [] = return []
+  replaceOwnParameters f = forM f $ \v -> replaceOwnParameters v
+
+instance (Parameterized a, Parameterized b) => Parameterized (a,b) where
+  flattenParameters (a,b) = flattenParameters a ++ flattenParameters b
+  replaceOwnParameters (a,b) = do
+    a' <- replaceOwnParameters a
+    b' <- replaceOwnParameters b
+    return (a',b')
+
+instance Parameterized Buffer where
+  flattenParameters _ = []
+  replaceOwnParameters = return
+
 instance Parameterized Double where
   flattenParameters _ = []
   replaceOwnParameters = return
 
-instance Parameterized [Int] where
+instance Parameterized Float where
```

Comment:
for this one I tried `instance (Scalar a) => Parameterized a`, but GHC felt that was not okay. :confused: